dashboard/modules/reporter/reporter_agent.py

```diff
@@ -506,19 +520,20 @@ def _get_workers(self):
                 if w.status() != psutil.STATUS_ZOMBIE
             ]
 
-    @staticmethod
-    def _get_raylet_proc():
+    def _get_raylet_proc(self):
         try:
-            curr_proc = psutil.Process()
-            # Here, parent is always raylet because the
-            # dashboard agent is a child of the raylet process.
-            parent = curr_proc.parent()
-            if parent is not None:
-                if parent.pid == 1:
+            if not self._raylet_proc:
+                curr_proc = psutil.Process()
+                # Here, parent is always raylet because the
+                # dashboard agent is a child of the raylet process.
+                self._raylet_proc = curr_proc.parent()
```

Comment:
this is necessary because psutil cpu_percent will return the CPU percent from the last time the method is called. The first invocation is always is 0. So we should cache the Process(). Note that the raylet pid won't change unless it is dead (which is unrecoverable).